### ISSUE TITLE
Issues with SVGFigure Width and Height not being set

### ISSUE THREAD

**Issue by toothrobber** [btel/svg_utils#73]
In the following code, I wanted to create an sag with a size of 1000 by 1000. Stepping through the code  the values of the width and the height were not set because they generated an exception in the Width and Height Setters. This resulted in the width and the height not being set.

Please note that I tried many combinations of values for width and height. 

    # create an empty image
    unit_fig = svgutils.transform.SVGFigure(width=1000.0, height=1000.0)

When the code reach the setter, the line self._width = value.value would fail.

    def width(self, value):
        **self._width = value.value**
        self.root.set("width", str(value))
        self.root.set("viewBox", "0 0 %s %s" % (self._width, self._height))

I replaced self._width = value. This fixed my problem.

robert

**Comment by hartwork** [comment]
I'd expect that to raise `AttributeError`which is handled at https://github.com/btel/svg_utils/blob/10a082008a40a387cd80ef044c3efe6a8aa79bd6/src/svgutils/transform.py#L244 . Which version of svgutils are you running?

**Comment by hartwork** [comment]
PS: Works for me with 0.3.4:

```console
# cd "$(mktemp -d)"
# virtualenv --python=python3.9 venv
# source venv/bin/activate
# pip install svgutils==0.3.4
# python -c $'import svgutils\nsvgutils.transform.SVGFigure(width=1000.0, height=1000.0)' && echo 'no exception'
no exception
```

**Comment by toothrobber** [comment]
Hello I used pip3 install svgutils

I believe it is version 0.3.4

How would I check?

**Comment by hartwork** [comment]
```console
# pip show svgutils | fgrep Version:
Version: 0.3.4
```

**Comment by toothrobber** [comment]
Hello Again

I took your example where no exception was generated and ran the following test. Use your code to generate a figure. Save that figure and look for the width and the height values in the svg file.

No Value for Width or Height are saved to the file. This indicated that the values for Width and Height are not being saved. 

~ % cat tmp.svg 
<?xml version='1.0' encoding='ASCII' standalone='yes'?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1"/>
~ % 

Python 3.9.2 (default, Mar 15 2021, 17:37:51) 
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
>>> import svgutils
>>> fig = svgutils.transform.SVGFigure(width=1000.0, height=1000.0)
>>> fig.save('tmp.svg')
>>> print(fig.width)
**None**
>>> print(fig.height)
**None**
>>> 

Please note I stepped through the code and the width and the height values are not saved.

Robert

**Comment by hartwork** [comment]
I confirm: the written SVG file does not carry width and height with svgutils 0.3.4. I guess #62 was not the full deal yet, then.

**Comment by toothrobber** [comment]
If you look at the code for the Width and the Height the "value.value" does not work.

ef width(self, value):
    **self._width = value.value**

**Comment by hartwork** [comment]
Yes, that is why `AttributeError` is caught. The code doesn't consider all cases yet, I suppose.

**Comment by cmahnke** [comment]
There is also an issue setting a value to a percentage, since the `Unit` class doesn't recognize it. But maybe it's another issue?
